packages/presence/src/channel.ts

```diff
@@ -56,19 +58,23 @@ export class Channel implements IChannel {
     }
   }
 
-  broadcast<T>(eventName: string, payload: T): void {
-    this.#broadcast(
-      eventName,
-      this.#getDataPacket({
-        state: {
-          id: this.#state.id,
-        },
-        payload,
-      })
-    );
+  broadcast<T extends JsonSerializable>(eventName: string, payload: T): void {
+    try {
+      const _payload = JsonSerializer.serialize(payload)
+
+      this.#broadcast(
+        eventName,
+        this.#getDataPacket({
+          payload: _payload
+        })
```

Comment:
```suggestion
        data: _payload
```

as the discussion: https://github.com/orgs/xileteam/discussions/672